# Notebook para el calculo de ratios financieras

In [2]:
#configuracion para poder importar desde src
import sys
sys.path.append("..")

%load_ext autoreload
%autoreload 2

import pandas as pd
from src.ratios import calcular_ratios

# Cargar el panel limpio que guardó 01_carga
panel = pd.read_csv("../data/processed/panel_2018_2024.csv")

# Calcular los ratios con la función del módulo
panel = calcular_ratios(panel)

panel[["razon_corriente", "endeudamiento", "apalancamiento", "roa",
       "margen_neto", "margen_operacional", "rotacion_activos"]].describe()

The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


,razon_corriente,endeudamiento,apalancamiento,roa,margen_neto,margen_operacional,rotacion_activos
count,2.008700e+04,20109.000000,2.010900e+04,20109.000000,1.869800e+04,18698.000000,19109.000000
mean,7.565369e+02,5.521519,-3.926324e+02,2.111328,-7.549749e+01,-55.624431,12.735478
std,7.104677e+04,619.978680,5.629688e+04,308.734924,7.577922e+03,5241.500949,1599.399428
min,0.000000e+00,0.000000,-7.983212e+06,-481.825106,-1.024338e+06,-703207.000000,0.000000
25%,1.068575e+00,0.327273,3.349713e-01,-0.008747,-5.862654e-03,0.014842,0.320849
50%,1.584456e+00,0.562716,1.000823e+00,0.021624,3.070125e-02,0.070542,0.888186
75%,2.762632e+00,0.787095,2.530569e+00,0.072524,9.095579e-02,0.154277,1.457927
max,8.990994e+06,87785.000000,7.291635e+03,43777.000000,8.290674e+03,7260.627604,221094.000000


Razón corriente: mediana 1.58, con el rango típico entre 1.07 y 2.76. Es normal, la mayoría de empresas tiene entre 1 y 2.7 veces sus pasivos corrientes en activos corrientes.
Endeudamiento: mediana 0.56 (financian el 56% de sus activos con deuda), rango 0.33–0.79. 
ROA: mediana 2.2%, rango −0.9% a 7.3%. Rentabilidades modestas pero realistas.
Margen neto: mediana 3.1%. Margen operacional: mediana 7.0%. Rotación de activos: mediana 0.89. Todos en rangos perfectamente sensatos.

Parecen haber grandes outliers